infrastructure/matrix/storer.go

```diff
@@ -118,7 +118,9 @@ func (s *storer) get(what string, id string) (string, error) {
 	var r row
 	result := s.db.Collection(table).Find(db.Cond{"bot_id": s.botID, "id": id, "what": what})
 	if err := result.One(&r); err != nil {
-		return "", err
+		if !errors.Is(err, db.ErrNoMoreRows) {
+			return "", err
```

Comment:
We don't want to return an error when row doesn't exist, the `!` check. For noMoreRows error, its going to return empty string only but with nil error.

Reply:
@psrpinto I am going to merge this but feel free to comment further if you have concerns / questions.

Reply:
When there are no results, we don't go in the `if` on line 121. I just tried it by debugging a simple test, and can confirm we end up on `return r.Value, nil` on line 126. Incidentally, `r.Value` is empty string, so the end result is the same.

Here's the test I used:

```go
func TestLoadMissingEntry(t *testing.T) {
	database.Test(func(session db.Session) {
		botID := uint64(1)
		userID := "@foo:matrix.test"
		storer := NewStorer(session, botID)

		got := storer.LoadFilterID(id.UserID(userID))
		if got != "" {
			// got is empty here because we returned r.Value, which was an empty string
		}
	})
}
```

Reply:
Yeah! That's exactly my intention with that `errors.Is()` check since we don't want that to be logged (its not really an error) even though upper treats it that way. And the end result is same, empty string. I am not seeing a reason this should be any different?

Reply:
Ok, I understand now. We return all errors except "not found". Please ignore, I got confused there at some point.